.github/workflows/update-api-files.yml

```diff
@@ -37,10 +37,14 @@ jobs:
           git config user.name "github-actions[bot]"
       # If run from a PR, checkout PR branch and merge in latest changes from main branch
       - name: Checkout PR branch and merge main
+        id: checkout-pr-branch
         if: ${{ github.event_name != 'workflow_dispatch' }}
-        run: |
-          git checkout ${{ github.event.pull_request.head.ref }}
-          git merge origin/main
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Merge main into PR branch
+        if: ${{ steps.checkout-pr-branch.outcome == 'success' }}
```

Comment:
Is this needed? I assume the whole action should halt if the outcome is not success?

Reply:
I was under the assumption that we want ot run through the other commands if nothing was checked out. 
Basing it off the comment
`      # If run from a PR, checkout PR branch and merge in latest changes from main branch
`

Reply:
If not, and we only want to continue with merging even if a branch was not checked out then we can remove the if statement here and the id too